VideoLocker/src/main/java/org/edx/mobile/module/notification/EdxParsePushBroadcastReceiver.java

```diff
@@ -32,101 +33,129 @@
     //UI and animation time need to get from production team.
     private boolean enableFlyinWindow = false;
 
-    public void onReceive(android.content.Context context, android.content.Intent intent) {
-        super.onReceive(context,intent);
-     }
+    public void onReceive(Context context, Intent intent) {
+        super.onReceive(context, intent);
+    }
 
 
-    protected void onPushReceive(android.content.Context context, android.content.Intent intent) {
-        try{
+    protected void onPushReceive(Context context, Intent intent) {
+        try {
             CourseUpdateNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
-            if ( payload != null && ParseHandleHelper.hasNotificationHash(context, payload.getIdentifier()) )
+            if (payload != null && ParseHandleHelper.hasNotificationHash(context, payload.getIdentifier()))
                 return;
-            if( enableFlyinWindow ) {
+            if (enableFlyinWindow) {
                 Intent intent2 = new Intent(context, NotificationFlyinService.class);
                 intent2.putExtra("com.parse.Data", intent.getExtras().getString("com.parse.Data"));
                 context.startService(intent2);
             }
-        }catch (Exception ex){
+        } catch (Exception ex) {
             logger.debug(ex.toString());
         }
         super.onPushReceive(context, intent);
     }
 
 
-    protected void onPushDismiss(android.content.Context context, android.content.Intent intent) {
+    protected void onPushDismiss(Context context, Intent intent) {
         super.onPushDismiss(context, intent);
     }
 
-    protected void onPushOpen(android.content.Context context, android.content.Intent intent) {
+    protected void onPushOpen(Context context, Intent intent) {
         super.onPushOpen(context, intent);
-        CourseUpdateNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
-         String courseId = payload == null ? "" : payload.getCourseId();
+        final CourseUpdateNotificationPayload payload = ParseHandleHelper.extractPayload(intent);
```

Comment:
The payload type and the generic logic here and in the `getNotifications()` method would need to be modified to be more generic if we are to support discussion notifications as well, but I guess that's a job for later.